### DISCUSSION TITLE
Missing Essentia with Gaia static builds

### DISCUSSION THREAD

**Discussion by regorxxx** [MTG/essentia#1279] [Q&A] [answered]
It has been months since it was reported that [Essentia + Gaia can not be built for windows following the current guide and source](https://github.com/MTG/essentia/issues/1265)

Neither the Acousticbrainz team, nor Essentia team seems to provide an answer to this problem. Can someone provide binaries? Binaries on web are outdated and missing Gaia.

**Comment by regorxxx** [comment]
**At this point I don't mind paying someone to do it, as long as it can be done. Being serious.**

I'm really surprised to see such bad support on an open source project, considering this is an ongoing problem since years.

**Comment by regorxxx** [comment]
Bump.

**Comment by wo80** [comment]
@regorxxx Please take a look at https://github.com/MTG/essentia/issues/1398. I'd be glad if you could test the cmake build and report any problems.

EDIT: note that there's also a cmake build script available for Gaia, but I haven't tested it since the old models seem to be unsupported: https://github.com/wo80/gaia/tree/cmake

**Reply by regorxxx** [reply]
As far as I have checked, that provides python wheels but not actual binaries ready to run as .exe, which is the thing missing for windows since years.

.exe files are pretty important for portable installs and sharing with other users in plugin environments.

**Reply by wo80** [reply]
I haven't released any binaries, except build artifacts of the `cmake-python` branch. If you need binaries, you could fork the project, modify the `build-cmake` workflow to build the examples and upload the desired exe as an artifact.

The project can be configured with option `BUILD_EXAMPLES=ON`, then all examples are built. Not sure which exe exactly you need...

In case you are still relying on the SVM models, the `streaming_extractor_music_svm` example should be built when enabling the option `USE_GAIA2=ON`. Obviously this has a dependency on Gaia and Qt.

I have enabled the issues section on my fork, so if you face any concrete problems building Essentia or Gaia, those can be discussed at https://github.com/wo80/essentia/issues

**Reply by regorxxx** [reply]
Ok, wil do so! Thank you

**Comment by regorxxx** [comment, marked as answer]
Well dev's support is non-existent but with someone-else help we managed to provided windows binaries:
https://github.com/wo80/essentia/issues/5

So anyone on Windows needing Essentia + Gaia binaries, they can be found here:

https://github.com/regorxxx/essentia/actions/workflows/build-win-binaries-gaia.yml

Click on latest workflow run and download the artifact (bottom)

<img width="1902" height="633" alt="Image" src="https://github.com/user-attachments/assets/5864243c-eb0a-49e8-a91a-be926c9f29f0" />

That build includes streaming_extractor_freesound.exe, streaming_extractor_music.exe and the needed dll.

On the actions page for the same repository you may also find Tensorflow builds or the standard build without any library (which includes standalone binaries to extract the key, bpm, etc.).

Alternatively, binaries + models + some extras may also be found here:
https://github.com/regorxxx/foobar2000-assets/tree/main/essentia-2.1b6

**Reply by brendanerofeev** [reply]
For anyone just finding this, the binaries in @regorxxx repo are expired (90 artifact expiry), so I've forked the repo and modified the workflows to create releases, which don't expire.  If you're in need you can find them here: https://github.com/brendanerofeev/essentia/releases

**Reply by regorxxx** [reply]
You don't need to clone the workflow, the comment you have replied includes a link with the all binaries + models and helpers. Both hosted at github and mega.
